modules/tensor_mechanics/src/materials/TrussMaterial.C

```diff
@@ -0,0 +1,106 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "TrussMaterial.h"
+#include "Material.h"
+#include "MooseMesh.h"
+#include "NonlinearSystem.h"
+
+//libmesh includes
+#include "libmesh/quadrature.h" 
+
+template<>
+InputParameters validParams<TrussMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addParam<std::string>("base_name", "Optional parameter that allows the user to define multiple mechanics material systems on the same block, i.e. for multiple phases");
+  params.addRequiredParam<std::vector<NonlinearVariableName> >("displacements", "The displacements appropriate for the simulation geometry and coordinate system");
+  params.addRequiredParam<Real>("youngs_modulus", "Young's modulus for truss element");
+  params.addCoupledVar("youngs_modulus_var","Variable containing Young's modulus");
+  return params;
+}
+
+TrussMaterial::TrussMaterial(const InputParameters & parameters) :
+    Material(parameters),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : "" ),
+    _my_youngs_modulus(isParamValid("youngs_modulus") ? getParam<Real>("youngs_modulus") : 0),
+    _youngs_modulus_coupled(isCoupled("youngs_modulus_var")),
+    _youngs_modulus_var(_youngs_modulus_coupled ? coupledValue("youngs_modulus_var"): _zero),  
+    _total_stretch(declareProperty<Real>(_base_name + "total_stretch")),
+    _elastic_stretch(declareProperty<Real>(_base_name + "elastic_stretch")),
+    _axial_stress(declareProperty<Real>(_base_name + "axial_stress")),
+    _e_over_l(declareProperty<Real>(_base_name + "e_over_l"))
+{
+  const std::vector<NonlinearVariableName> & nl_vnames(getParam<std::vector<NonlinearVariableName> >("displacements"));
+  _ndisp = nl_vnames.size();
+
+  // fetch nonlinear variables
+  for (unsigned int i = 0; i < _ndisp; ++i)
+    _disp_var.push_back(&_fe_problem.getVariable(_tid, nl_vnames[i]));
+
+  if (parameters.isParamValid("youngs_modulus"))
+  {
+    if (_youngs_modulus_coupled)
+      mooseError("Cannot specify both youngs_modulus and youngs_modulus_var");
+  }
+  else
+  {
+    if (!_youngs_modulus_coupled)
+      mooseError("Must specify either youngs_modulus or youngs_modulus_var");
+  }
```

Comment:
All this logic is not necessary as MOOSE provides default coupling, which let's you specify a number in place of a variable name! Remove `youngs_modulus` and rename `youngs_modulus_var` to `youngs_modulus`!